src/main/java/uk/gov/digital/ho/hocs/audit/service/AuditEventService.java

```diff
@@ -24,76 +27,57 @@ public class AuditEventService {
-        log.debug("Case {} setting deleted to {}", caseUUID, deleted);
         List<AuditEvent> audits = auditRepository.findAuditDataByCaseUUID(caseUUID);
         for (AuditEvent audit : audits) {
             audit.setDeleted(deleted);
             auditRepository.save(audit);
         }
-        log.info("Case {} set {} audits deleted to {}", caseUUID, audits.size(), deleted);
+        log.info("Set Deleted=({}) for {} audit lines for caseUUID: {}", deleted, audits.size(), caseUUID, value(EVENT, AUDIT_EVENT_DELETED));
         return audits.size();
     }
 
     @Transactional(readOnly = true)
-    public List<AuditEvent> getAuditDataByCaseUUID(UUID caseUUID, String types) {
+    public GetAuditListResponse getAuditDataByCaseUUID(UUID caseUUID, String[] filterTypes) {
         log.debug("Requesting Audit for Case UUID: {} ", caseUUID);
-        String[] filterTypes = types.split(",");
-        return auditRepository.findAuditDataByCaseUUIDAndTypesIn(caseUUID, filterTypes);
+        var auditResponses = auditRepository.findAuditDataByCaseUUIDAndTypesIn(caseUUID, filterTypes);
+        return buildAuditListResponse(auditResponses);
     }
 
-    public List<AuditEvent> getAuditDataByCaseUUID(UUID caseUUID, String types, LocalDate from) {
-        log.debug("Requesting Audit for Case UUID: {} ", caseUUID);
-        String[] filterTypes = types.split(",");
-        return auditRepository.findAuditDataByCaseUUIDAndTypesInAndFrom(caseUUID, filterTypes, from);
+    @Transactional(readOnly = true)
+    public GetAuditListResponse getAuditDataByCaseUUID(UUID caseUUID, String[] filterTypes, LocalDate fromDate) {
+        log.debug("Requesting Audit for Case UUID: {} FromDate: {}", caseUUID, fromDate);
+        var auditResponses = auditRepository.findAuditDataByCaseUUIDAndTypesInAndFrom(caseUUID, filterTypes, fromDate);
+        return buildAuditListResponse(auditResponses);
     }
 
-    private static void validateNotNull(AuditEvent auditEvent) {
-        String correlationID = auditEvent.getCorrelationID();
-        String raisingService = auditEvent.getRaisingService();
-        String namespace = auditEvent.getNamespace();
-        LocalDateTime auditTimestamp = auditEvent.getAuditTimestamp();
-        String type = auditEvent.getType();
-        String userID = auditEvent.getUserID();
-
-        if (correlationID == null || raisingService == null || namespace == null || auditTimestamp == null || type == null || userID == null) {
-            throw new EntityCreationException("Cannot create Audit - null input(%s, %s, %s, %s, %s, %s, %s)",
-                    correlationID,
-                    raisingService,
-                    auditEvent.getAuditPayload(),
-                    namespace,
-                    auditTimestamp,
-                    type,
-                    userID);
-        }
+    private GetAuditListResponse buildAuditListResponse(Stream<AuditEvent> auditEvents) {
```

Comment:
This feels cursed, not sure I agree this should be the place for this. Not sure the detaching benefit outways the clean code.

Reply:
this is what I want to test :D. This service doesn't have the GC profile of other services, but i'd like to establish the pattern. I introduced the `from` pattern but I don't feel passing the EntityManager into that is right.